plugins/src/android/OptimoveSDKPlugin.java

```diff
@@ -48,11 +87,154 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
+        JSONObject inboxItem = args.optJSONObject(0);
+        int messageId = -1;
+        if (inboxItem != null) {
+            messageId = args.optInt(0);
+        }
+        if (inboxItem == null || messageId == -1) {
+            callbackContext.error("Message not found or not available");
+            return;
+        }
+
+        List<InAppInboxItem> items = OptimoveInApp.getInstance().getInboxItems();
+        for (InAppInboxItem item : items) {
+            if (item.getId() == messageId) {
+                boolean result = OptimoveInApp.getInstance().deleteMessageFromInbox(item);
+
+                if (result) {
+                    callbackContext.success();
+                    return;
+                }
+
+                break;
+            }
+        }
+
+        callbackContext.error("Message not found or not available");
+    }
+
+    private void inAppPresentInboxMessage(JSONArray args, CallbackContext callbackContext) {
+        JSONObject inboxItem = args.optJSONObject(0);
+        int messageId = -1;
+        if (inboxItem != null) {
+            messageId = args.optInt(0);
+        }
+        if (inboxItem == null || messageId == -1) {
+            callbackContext.error("Message not found or not available");
+            return;
+        }
+
+        List<InAppInboxItem> items = OptimoveInApp.getInstance().getInboxItems();
+        for (InAppInboxItem item : items) {
+            if (item.getId() == messageId) {
+                OptimoveInApp.InboxMessagePresentationResult result = OptimoveInApp.getInstance()
+                        .presentInboxMessage(item);
+
+                if (result == OptimoveInApp.InboxMessagePresentationResult.PRESENTED) {
+                    callbackContext.success();
+                    return;
+                } else {
+                    break;
+                }
+            }
+        }
+
+        callbackContext.error("Message not found or not available");
+    }
+
+    private void inAppGetInboxSummary(CallbackContext callbackContext) {
+        OptimoveInApp.getInstance().getInboxSummaryAsync((InAppInboxSummary summary) -> {
+            if (summary == null) {
+                callbackContext.error("Could not get inbox summary");
+                return;
+            }
+
+            try {
+                JSONObject res = new JSONObject();
+                res.put("totalCount", summary.getTotalCount());
+                res.put("unreadCount", summary.getUnreadCount());
+                callbackContext.success(res);
+            } catch (JSONException e) {
+                e.printStackTrace();
+                callbackContext.error(e.getMessage());
+            }
+        });
+    }
+
+    private void inAppMarkAsRead(JSONArray args, CallbackContext callbackContext) {
+        try {
+            JSONObject inboxItem = args.optJSONObject(0);
+            int messageId = -1;
+            if (inboxItem != null) {
+                messageId = args.optInt(0);
```

Comment:
here i dont get these conditionals a bit. We take 1st argument json object. If it is not null, we take 1st argument int?

is intention allow passing either id and inboxItem?

Reply:
it should be " messageId = inboxItem.optInt("id");"
I fixed it now

Reply:
updated to pass id into javaland